docs/topics/debug-flow-with-idea.md

```diff
@@ -0,0 +1,126 @@
+
+2. Open the `main.kt` file in `src/main/kotlin`.
+
+    The `src` directory contains Kotlin source files and resources. The `main.kt` file contains sample code that will print `Hello World!`.
+
+2. Create the `simple()` function that returns a flow of three numbers:
+
+    * Use the [`delay()`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/delay.html) function to imitate CPU-consuming blocking code. It suspends the coroutine for 100 ms without blocking the thread.
+    * Produce the values in the `for` loop using the [`emit()`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/-flow-collector/emit.html) function.
+
+    ```kotlin
+    import kotlinx.coroutines.*
+    import kotlinx.coroutines.flow.*
+    import kotlin.system.*
+ 
+    fun simple(): Flow<Int> = flow {
+        for (i in 1..3) {
+            delay(100)
+            emit(i)
+        }
+    }
+    ```
+
+3. Change the code in the `main()` function:
+
+    * Use the [`runBlocking()`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/run-blocking.html) block to wrap a coroutine.
+    * Collect the emitted values using the [`collect()`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.flow/collect.html) function.
+    * Use the [`delay()`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/delay.html) function to imitate CPU-consuming code. It suspends the coroutine for 300 ms without blocking the thread.
+    * Print the collected value from the flow using the [`println()`](https://kotlinlang.org/api/latest/jvm/stdlib/stdlib/kotlin.io/println.html) function.
+
+    ```kotlin
+    fun main() = runBlocking {
+        simple()
+            .collect { value ->
+                delay(300)
+                println(value)
+            }
+    }
+    ```
+
+4. Build the code by clicking **Build Project**.
+
+    ![Build an application](flow-build-project.png)
+
+## Debug the coroutine
+
+1. Set a breakpoint at the at the line where the `emit()` function is called:
+
+    ![Build a console application](flow-breakpoint.png)
+
+2. Run the code in debug mode by clicking **Debug** next to the run configuration at the top of the screen.
+
+    ![Build a console application](flow-debug-project.png)
+
+    The **Debug** tool window appears: 
+    * The **Frames** tab contains the call stack.
+    * The **Variables** tab contains variables in the current context. It tells us that the flow is emitting the first value.
+    * The **Coroutines** tab contains information on running or suspended coroutines.
+
+    ![Debug the coroutine](flow-debug-1.png)
+
+3. Resume the debugger session by clicking **Resume program** in the **Debug** tool window. The program stops at the same breakpoint.
+
+    ![Debug the coroutine](flow-resume-debug.png)
```

Comment:
These links are broken for MD files. Is it expected?

Reply:
Yep, the WebHelp file structure ignores hierarchy, it's flat.

Reply:
We discussed this in Slack last year. It's expected for now. We'll think what we can do with this in the future if we see that this is a real issue for users. 
Based on current information, it doesn't seem to be an issue for contributors. KMM docs have much more images than coroutines docs and we didn't get any feedback from users complaining that images were not rendered in MD.

If we get such feedback, we'll reprioritize this task.